src/openforms/contrib/kadaster/api/views.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+from django.contrib.gis.geos import fromstr
+from django.utils.translation import gettext_lazy as _
+
+import requests
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.views import APIView
+from zds_client import ClientError
+
+from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
+from openforms.api.views.mixins import ListMixin
+from openforms.contrib.kadaster.models import KadasterApiConfig
+from openforms.submissions.api.permissions import AnyActiveSubmissionPermission
+
+from .serializers import MapSearchSerializer
+
+logger = logging.getLogger(__name__)
+
+
+@extend_schema(
+    summary=_("List BAG address suggestions."),
+    parameters=[
+        OpenApiParameter(
+            "q",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("The search query we send to the pdok locatieserver api."),
+            required=True,
+        )
+    ],
+    responses={
+        200: MapSearchSerializer(many=True),
+        status.HTTP_400_BAD_REQUEST: ValidationErrorSerializer,
+        status.HTTP_403_FORBIDDEN: ExceptionSerializer,
+    },
+)
+class MapSearchView(ListMixin, APIView):
+    authentication_classes = ()
+    permission_classes = [AnyActiveSubmissionPermission]
+    serializer_class = MapSearchSerializer
+
+    def get_objects(self):
+        config = KadasterApiConfig.get_solo()
+        if not config.kadaster_service:
+            raise ValidationError(
+                {"invalid_params": _("No service defined for the Kadaster service.")}
+            )
+
+        query = self.request.GET.get("q")
+        if not query:
+            raise ValidationError({"invalid_params": _("Missing query parameter 'q'")})
+
+        kadaster_client = config.kadaster_service.build_client()
+        query_params = {"q": f"{query}"}
+
+        bag_response = dict()
+        try:
+            bag_response = kadaster_client.operation(
+                operation_id="free",
+                data=None,
+                url="v3_1/free",
+                method="GET",
+                request_kwargs=dict(
+                    params=query_params,
+                ),
+            )
+        except (ClientError, requests.RequestException):
+            logger.exception("couldn't retrieve pdok locatieserver data")
```

Comment:
```suggestion
        try:
            bag_response = kadaster_client.operation(
                operation_id="free",
                data=None,
                url="v3_1/free",
                method="GET",
                request_kwargs=dict(
                    params=query_params,
                ),
            )
        except (ClientError, requests.RequestException):
            bag_response = {}
            logger.exception("Couldn't retrieve PDOK locatieserver data")
```
I'm not a fan of re-assigning variables, it's error-prone when more/other cases need to be handled.